python3.10.8 环境下尝试格式化12小时制报错

08-24 20:01:23 [ERROR] nonebot | Running Matcher(type='message', module=nonebot_plugin_heweather) failed.
Traceback (most recent call last):
  File "<string>", line 17, in <module>
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\__init__.py", line 333, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 199, in run
    uvicorn.run(
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\uvicorn\main.py", line 587, in run
    server.run()
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1899, in _run_once
    handle._run()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\message.py", line 467, in check_and_run_matcher
    await _run_matcher(
> File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\message.py", line 419, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 759, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 734, in simple_run
    await handler(
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot\dependencies\__init__.py", line 108, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot_plugin_heweather\__init__.py", line 51, in _
    img = await render(w_data)
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot_plugin_heweather\render_pic.py", line 33, in render
    "hours": add_hour_data(weather.hourly.hourly),
  File "C:\Users\Administrator\ATRI\.venv\lib\site-packages\nonebot_plugin_heweather\render_pic.py", line 48, in add_hour_data
    hour.hour = date_time.strftime("%-I%p")
ValueError: Invalid format string

08-25 11:41:44 [ERROR] nonebot | Running Matcher(type='message', module=extensive_plugin.nonebot_plugin_heweather) failed.
Traceback (most recent call last):
  File "D:\BOT\zhenxun_bot\bot.py", line 22, in <module>
    nonebot.run()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\__init__.py", line 309, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\drivers\fastapi.py", line 198, in run
    uvicorn.run(
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\uvicorn\main.py", line 578, in run
    server.run()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1906, in _run_once
    handle._run()
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\message.py", line 467, in check_and_run_matcher
    await _run_matcher(
 File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\message.py", line 419, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\internal\matcher\matcher.py", line 753, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\internal\matcher\matcher.py", line 728, in simple_run
    await handler(
  File "C:\Users\550N\AppData\Local\Programs\Python\Python310\lib\site-packages\nonebot\dependencies\__init__.py", line 108, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "D:\BOT\zhenxun_bot\extensive_plugin\nonebot_plugin_heweather\__init__.py", line 52, in _
    await w_data.load_data()
  File "D:\BOT\zhenxun_bot\extensive_plugin\nonebot_plugin_heweather\weather_data.py", line 68, in load_data
    ) = await asyncio.gather(
  File "D:\BOT\zhenxun_bot\extensive_plugin\nonebot_plugin_heweather\weather_data.py", line 157, in _hourly
    return HourlyApi(**res.json())
  File "pydantic\main.py", line 342, in pydantic.main.BaseModel.__init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for HourlyApi
hourly
  field required (type=value_error.missing)

我也报错了

--- a/nonebot_plugin_heweather/render_pic.py
+++ b/nonebot_plugin_heweather/render_pic.py
@@ -1,3 +1,4 @@
+import platform
 from datetime import datetime
 from typing import List
 from pathlib import Path
@@ -44,8 +45,11 @@ def add_hour_data(hourly: List[Hourly]):
     high = max([int(hour.temp) for hour in hourly])
     low = int(min_temp - (high - min_temp))
     for hour in hourly:
-        date_time = datetime.fromisoformat(hour.fxTime)  # 2023-08-09 23:00:00+08:00
-        hour.hour = date_time.strftime("%H:%M")
+        date_time = datetime.fromisoformat(hour.fxTime)
+        if platform.system() == "Windows":
+            hour.hour = date_time.strftime("%#I%p")
+        else:
+            hour.hour = date_time.strftime("%-I%p")
         hour.temp_percent = f"{int((int(hour.temp) - low) / (high - low) * 100)}px"
     if QWEATHER_HOURLYTYPE == HourlyType.current_12h:
         hourly = hourly[:12]

